ansible/roles/kubernetes-common/templates/etc/systemd/system/kubelet.service.d/09-extra-args.conf

```diff
@@ -0,0 +1,2 @@
+[Service]
+Environment="KUBELET_EXTRA_ARGS={% if kubernetes_common.primary_interface is defined %} --node-ip={{kubernetes_node_ip}}{% endif %}{% for k, v in kubernetes_common.get('kubelet_extra_args', {}).items() %} --{{k}}='{{v}}'{%- endfor %}"
```

Comment:
is it possible join `kubernetes_common.primary_interface` into `kubelet_extra_args` dict before executing this file? Then you could ditch the `{% if %}` block and always do the for loop. 

Might be more trouble than it's worth but this is not a very pretty template.

Reply:
Unfortunately, I thought of that and it is not really possible.  I agree that this is a very very ugly template, but this is an artifact of the way that the kubelet packages implement systemd unit files.